src/consensus/pbft_manager.cpp

```diff
@@ -1638,6 +1657,19 @@ bool PbftManager::giveUpSoftVotedBlock_() {
   unsigned long elapsed_wait_soft_voted_block_in_ms =
       std::chrono::duration_cast<std::chrono::milliseconds>(soft_voted_block_wait_duration).count();
 
+  auto pbft_block = pbft_chain_->getUnverifiedPbftBlock(previous_round_next_voted_value_);
+  if (!pbft_block) {
+    pbft_block = db_->getPbftCertVotedBlock(previous_round_next_voted_value_);
+  }
+
+  if (pbft_block) {
+    // Have a block, but is it valid?
+    if (!checkPbftBlockValid_(previous_round_next_voted_value_)) {
+      // Received the block, but not valid
+      return true;
+    }
```

Comment:
Do we need add:
else {
    return false;
}

If receive the block already and the block is valid, should continue with the value. Don't terminate

Reply:
@gaoqi8649  I thought so at first too, but being valid in this sense (ie a well signed block) isn't sufficient because it can be a malicious block that points to a bogus DAG anchor that doesn't exist.  In that case we DO want to give up.